modules/platforms/dotnet/Apache.Ignite.Core/Impl/Cache/CacheImpl.cs

```diff
@@ -949,6 +1243,45 @@ private Task<long> SizeAsync0(int? part, params CachePeekMode[] modes)
             });
         }
 
+        /// <summary>
+        /// Encodes peek modes, includes native near check.
+        /// </summary>
+        private int EncodePeekModes(bool loc, int? part, CachePeekMode[] modes, out bool onlyNativeNear, out int size)
+        {
+            size = 0;
+            onlyNativeNear = false;
+            
+            bool hasPlatformNear;
+            var modes0 = IgniteUtils.EncodePeekModes(modes, out hasPlatformNear);
+
+            if (hasPlatformNear)
+            {
+                if (!loc)
+                {
+                    throw new InvalidOperationException(
```

Comment:
It feels that Peek.All should work as "all available" instead of throwing an explicit exception.
Please, check tests, we could convert this into a warning temporary.

Reply:
Agree, fixed. No exceptions now.